grakn-graql-shell/src/main/java/ai/grakn/graql/JsonSession.java

```diff
@@ -52,7 +52,11 @@
 
     private final Session session;
 
-    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(runnable -> {
+        Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+        thread.setDaemon(true);
+        return thread;
+    });
```

Comment:
I've added a test now that confirms the number of non-daemon threads does not change. I'm concerned it may be a source of random failures, but we'll have to see.